framework/auth/cas.py

```diff
@@ -160,17 +160,18 @@ def _parse_profile(self, raw, access_token):
 
     def revoke_application_tokens(self, client_id, client_secret):
         """Revoke all tokens associated with a given CAS client_id"""
-        url = self.get_application_revocation_url()
-        data = {'client_id': client_id,
-                'client_secret': client_secret}
+        return self.revoke_tokens(payload={'client_id': client_id, 'client_secret': client_secret})
 
-        resp = requests.post(url, data=data)
+    def revoke_tokens(self, payload):
```

Comment:
It's not internal, it's intended to generalize revoking tokens given by cas (so that we don't need individual revocation methods for DevApps, PATs, and anything added in the future). It would be called in the `.deactivate` methods of those objects. `ApiOauth2Application` was not updated yet because it's out of scope of this PR (and would break at least one PR, #4441, that's currently open).